Sources/Fuzzilli/FuzzIL/JsOperations.swift

```diff
@@ -1047,21 +1047,37 @@ final class TestIn: JsOperation {
 
 }
 
+enum ParameterPattern {
+    case identifier
+    case object(properties: [ObjectPatternProperty])
+    case array(elements: [ParameterPattern])
+}
+
+struct ObjectPatternProperty {
+    let key: String
+    let value: ParameterPattern
+}
+
 // The parameters of a FuzzIL subroutine.
 public struct Parameters {
-    /// The total number of parameters.
     private let numParameters: UInt32
-    /// Whether the last parameter is a rest parameter.
     let hasRestParameter: Bool
 
-    /// The total number of parameters. This is equivalent to the number of inner outputs produced from the parameters.
     var count: Int {
         return Int(numParameters)
     }
 
-    init(count: Int, hasRestParameter: Bool = false) {
+    var objectPropertyNames: [[String]]
```

Comment:
Is this still used?

Reply:
Good catch, thanks for pointing that out. That is an artifact of a previous version where I did not have the enum and instead only saved indices of braces, which was not very maintainable/intuitive.